crates/db_views/src/comment/comment_view.rs

```diff
@@ -269,38 +285,32 @@ impl CommentQuery<'_> {
       // If a max depth is given, then you know its a tree fetch, and limits should be ignored
       // TODO a kludge to prevent attacks. Limit comments to 300 for now.
       // (i64::MAX, 0)
-      (300, 0)
+      300
     } else {
       // limit_and_offset_unlimited(o.page, o.limit)
-      limit_and_offset(o.page, o.limit)?
+      limit_fetch(o.limit)?
     };
+    query = query.limit(limit);
+
+    // Only sort by ascending for Old
+    let sort = o.sort.unwrap_or(CommentSortType::Hot);
+    let sort_direction = asc_if(sort == CommentSortType::Old);
```

Comment:
This will cause distinguished comments to go to the bottom when sorting by old. Either use ReverseTimestampKey here again, conditionally invert the distinguished key (using a new wrapper similar to ReverseTimestampKey), or don't sort by distinguished at all when using the Old sort type (I think this is the best solution, and maybe it also should be done for New sort for consistency). Update indexes to match.

Reply:
K I got this one handled (indexes I'll go over shortly.)